### PR TITLE
github: add release-checklist template

### LINKED ISSUE #176: docs/release-guide: move to checklist ticket template
On each release we publish a `.crate` file to crates.io, which is then used by distribution to package coreos-metadata. Its SHA256 is recorded at publish time in the index, eg 
https://github.com/rust-lang/crates.io-index/commit/e0bc5cf11001a1e79521f3fb21d568d2c50fed42.

In order to improve the integrity chain and to make manual checks easier, we should also record the SHA256 of the `.crate` file on local filesystem (before upload) and publish it in the release notes.

As per discussion with @ashcrow.

### PR DESCRIPTION
**Pull request by lucab**
This adds a bug template for GitHub detailing all steps that are
performed in the release process, as a checklist.
It replaces the previous "release guide" doc.

Closes: https://github.com/coreos/afterburn/issues/176

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+This guide requires:
+
+ * a web browser (and network connectivity)
+ * `git`
+ * GPG setup and personal key for signing
+ * `cargo` (suggested: latest stable toolchain from [rustup][rustup])
+ * `cargo-release` (suggested: `cargo install -f cargo-release`)
+ * A verified account on crates.io
+ * Write access to this GitHub project
+ * Upload access to this GitHub project
```
Comment (jlebon) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: That should be crates.io I'm guessing?

On that subject, is there value in publishing there?

**Review by jlebon — COMMENTED** [review]
Looks sane overall from a quick scan. Just some comments.

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (lucab) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: 1) good catch, we actually need both crates.io and github
2) packaging rules (surely Fedora, I think RH too) use the crate tarball from crates.io

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Release checklist
+
+These steps show how to release version `x.y.z` on the `origin` remote (this can be checked via `git remote -av`).
+Push access to the upstream repository is required in order to publish the new tag and the PR branch.
+
+- [ ] make sure the project is clean and prepare the environment:
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: The high-level checkboxes are redundant; maybe drop them?

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+---
+name: Release a new version
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: In the Ignition repo we omit this header, which prevents the release checklist from showing up as an option in the new-bug flow.  We then put a [direct link](https://github.com/coreos/ignition/issues/new?labels=kind/release&template=release-checklist.md) in the developer docs.

**Review by bgilbert — COMMENTED** [review]
Thanks for doing this!  :+1:

**Comment by lucab** [comment]
@jlebon @bgilbert I've updated this with your suggestions, PTAL (and thanks for reviewing).

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  - [ ] publish release
+
+- clean up the local environment:
+  - [ ] `unset RELEASE_VER`
+  - [ ] `unset UPSTREAM_REMOTE`
+  - [ ] `cargo clean`
+  - [ ] `rm -rf vendor`
+  - [ ] `git checkout master`
+  - [ ] `git pull ${UPSTREAM_REMOTE} master`
+  - [ ] `git push ${UPSTREAM_REMOTE} :release-${RELEASE_VER}`
```
Comment (ashcrow) [inline .github/ISSUE_TEMPLATE/release-checklist.md:74]: Technically not local environment, but it's fine :-)

**Review by ashcrow — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 4 above)
Comment (lucab) [reply .github/ISSUE_TEMPLATE/release-checklist.md:74]: Ah, right. Better suggestions? "clean up environment and leftovers"?

Comment (ashcrow) [reply .github/ISSUE_TEMPLATE/release-checklist.md:74]: Stating "Clean Up" would work IMHO.

Comment (lucab) [reply .github/ISSUE_TEMPLATE/release-checklist.md:74]: Ack, applied that.

**Review by ashcrow — APPROVED** [review]

**Review by bgilbert — APPROVED** [review]
LGTM!

**Pull request merged**